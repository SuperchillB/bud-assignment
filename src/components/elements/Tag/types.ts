import { DisplayProps, SpaceProps, WidthProps } from 'styled-system';

export type TagProps = {
  children: React.ReactNode;
  variant: 'success' | 'warning' | 'secondary';
} & SpaceProps &
  WidthProps &
  DisplayProps;
// } & TagFunctionsProps;

// export type TagFunctionsProps = TextProps & FlexProps;
