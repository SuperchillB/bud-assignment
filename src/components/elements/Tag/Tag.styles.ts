import styled from 'styled-components';
import { variant } from 'styled-system';
import { Box } from '../Box';

const TagContainer = styled(Box)(
  {
    textTransform: 'uppercase',
    textAlign: 'center',
  },
  variant({
    variants: {
      secondary: {
        color: 'primary',
        bg: 'bluePillBg',
      },
      loading: {
        color: 'primary',
        bg: 'loadingGrey',
      },
      // Could add more variants here for different types of Tags
    },
  }),
);

export default TagContainer;
