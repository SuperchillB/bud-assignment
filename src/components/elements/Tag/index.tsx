import { Paragraph } from '../Paragraph';
import { TagProps } from './types';
import TagContainer from './Tag.styles';

const Tag = ({ children, ...props }: TagProps) => (
  <TagContainer borderRadius="tag" display="inline-block" px="small" {...props}>
    <Paragraph my="tiny" variant="subTitle">
      {children}
    </Paragraph>
  </TagContainer>
);

export default Tag;
