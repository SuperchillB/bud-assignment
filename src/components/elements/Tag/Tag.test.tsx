import { render, screen as screenRTL } from '@testing-library/react';
import Tag from '.';

describe('Tag component', () => {
  test('renders its children', () => {
    const children = 'Tag test copy';
    render(<Tag variant="secondary">{children}</Tag>);

    expect(screenRTL.getByText(children)).toBeInTheDocument();
  });

  test('renders with given props and "secondary" variant', () => {
    const { container } = render(
      <Tag display={['none', 'none', 'inline-block']} variant="secondary">
        Test tag
      </Tag>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
