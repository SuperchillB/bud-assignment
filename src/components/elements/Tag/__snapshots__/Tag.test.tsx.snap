// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component renders with given props and "secondary" variant 1`] = `
<div
  class="sc-gsnTZi sc-dkzDqf jhAtse hoZrpZ"
  display="none,none,inline-block"
>
  <p
    class="sc-bczRLJ fQxWBD"
  >
    Test tag
  </p>
</div>
`;
